requests.go

```diff
@@ -111,13 +107,9 @@ func (p *RequestProtocol) requestShardPeer(
 
 	shardPeers := make(map[ShardIDType][]peer.ID)
 	for shardID, peers := range res.ShardPeers {
-		peerIDs := []peer.ID{}
-		for _, peerString := range peers.Peers {
-			peerID, err := peer.IDB58Decode(peerString)
-			if err != nil {
-				return nil, fmt.Errorf("error occurred when parsing peerID: %s", peerString)
-			}
-			peerIDs = append(peerIDs, peerID)
+		peerIDs, err := pbPeersToPeerIDs(peers)
+		if err != nil {
+			return nil, fmt.Errorf("peerID in wrong format: %v", err)
```

Comment:
Nitpick: what do you think about error message: `Failed to convert peer ID from PB string to peerID format, err: %v", err`?